### PR TITLE
Core: Boot Vitest runner when hovering the Testing Module

### PR DESCRIPTION
**Pull request by ghengeveld**
Adds a new `TESTING_MODULE_BOOT_REQUEST` event which is emitted when hovering the Testing Module. The Vitest manager is booted when this event fires, so that it's ready when a test run is started. This saves about 1.5-2 seconds on my Macbook Pro M1.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createSize |  0 B | 0 B | 0 B | - | - |
| generateSize |  77.7 MB | 77.7 MB | 143 B | **-4.35** | 0% |
| initSize |  146 MB | 146 MB | 1.32 kB | **-4.32** | 0% |
| diffSize |  68.4 MB | 68.4 MB | 1.17 kB | **3.78** | 0% |
| buildSize |  7.1 MB | 7.1 MB | 623 B | **4.97** | 0% |
| buildSbAddonsSize |  1.79 MB | 1.79 MB | 158 B | **1.43** | 0% |
| buildSbCommonSize |  195 kB | 195 kB | 0 B | - | 0% |
| buildSbManagerSize |  1.85 MB | 1.85 MB | 328 B | 0.74 | 0% |
| buildSbPreviewSize |  271 kB | 271 kB | 137 B | **21.7** | 0.1% |
| buildStaticSize |  0 B | 0 B | 0 B | - | - |
| buildPrebuildSize |  4.1 MB | 4.1 MB | 623 B | **4.14** | 0% |
| buildPreviewSize |  3 MB | 3 MB | 0 B | 0.82 | 0% |
| testBuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbAddonsSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbCommonSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbManagerSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbPreviewSize |  0 B | 0 B | 0 B | - | - |
| testBuildStaticSize |  0 B | 0 B | 0 B | - | - |
| testBuildPrebuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildPreviewSize |  0 B | 0 B | 0 B | - | - |

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createTime |  7.3s | 7.3s | -39ms | -1.2 | -0.5% |
| generateTime |  25s | 25s | 5ms | 0.74 | 0% |
| initTime |  16.9s | 15.9s | -1s -42ms | 0.24 | -6.6% |
| buildTime |  8.6s | 8.2s | -317ms | -0.93 | -3.8% |
| testBuildTime |  0ms | 0ms | 0ms | - | - |
| devPreviewResponsive |  5.8s | 6.5s | 755ms | 0.25 | 11.5% |
| devManagerResponsive |  4s | 4.3s | 292ms | 0.32 | 6.8% |
| devManagerHeaderVisible |  520ms | 617ms | 97ms | -0.09 | 15.7% |
| devManagerIndexVisible |  550ms | 653ms | 103ms | -0.12 | 15.8% |
| devStoryVisibleUncached |  841ms | 1.1s | 278ms | 0.55 | 24.8% |
| devStoryVisible |  549ms | 644ms | 95ms | -0.16 | 14.8% |
| devAutodocsVisible |  470ms | 524ms | 54ms | -0.47 | 10.3% |
| devMDXVisible |  487ms | 528ms | 41ms | -0.34 | 7.8% |
| buildManagerHeaderVisible |  535ms | 582ms | 47ms | -0.34 | 8.1% |
| buildManagerIndexVisible |  563ms | 601ms | 38ms | -0.57 | 6.3% |
| buildStoryVisible |  518ms | 569ms | 51ms | -1.04 | 9% |
| buildAutodocsVisible |  467ms | 479ms | 12ms | -1.05 | 2.5% |
| buildMDXVisible |  461ms | 464ms | 3ms | -0.99 | 0.6% |

## Greptile Summary

This pull request introduces a new event `TESTING_MODULE_BOOT_REQUEST` to improve the performance of the Vitest runner in Storybook. The key changes include:

- Added `TESTING_MODULE_BOOT_REQUEST` event to `code/core/src/core-events/index.ts`
- Implemented event listener for `TESTING_MODULE_BOOT_REQUEST` in `code/addons/test/src/preset.ts`
- Integrated `bootTestProviders` function in `SidebarBottom` component to emit the new event on hover
- Updated `code/core/src/manager/globals/exports.ts` to include the new event in globalized dependencies
- Modified `execute` function in test preset to accept an optional `eventName` parameter

These changes aim to preemptively boot the Vitest runner when hovering over the Testing Module, potentially saving 1.5-2 seconds of loading time.

### CODE REVIEW

**Review by valentinpalkovic — APPROVED** [review]
Rebased, since tests were failing. Otherwise, LGTM

File: code/core/src/manager/components/sidebar/SidebarBottom.tsx [hunk 1]
```diff
@@ -113,6 +114,8 @@ export const SidebarBottomBase = ({ api, notifications = [], status = {} }: Side
   const hasWarnings = warnings.length > 0;
   const hasErrors = errors.length > 0;
 
+  const bootTestProviders = useCallback(() => api.emit(TESTING_MODULE_BOOT_REQUEST), [api]);
```
Comment (valentinpalkovic) [inline code/core/src/manager/components/sidebar/SidebarBottom.tsx:117]: This event shouldn't be sent on subsequent hovers

**Review by valentinpalkovic — CHANGES_REQUESTED** [review]

File: code/core/src/manager/components/sidebar/SidebarBottom.tsx (hunk 1 above)
Comment (ndelangen) [reply code/core/src/manager/components/sidebar/SidebarBottom.tsx:117]: You're going to have to deal with the event arriving multiple times anyway.

The user might open multiple browsers or refresh, etc.

I'll leave it up to you to decide if adding a check here is worth the additional work/complexity)

Comment (ghengeveld) [reply code/core/src/manager/components/sidebar/SidebarBottom.tsx:117]: Why not? Providers may want to shut down after a while. They should already take care not to boot more than once.

**Review by valentinpalkovic — APPROVED** [review]

**Comment by ghengeveld** [comment]
Going to pause this PR for the time being.

**Pull request closed without merging**